### ISSUE TITLE
Wrong error highlight with `#[inherent]` proc-macro

### ISSUE THREAD

**Issue by jendrikw** [intellij-rust/intellij-rust#8350]
## Environment

* **IntelliJ Rust plugin version:** 0.4.162.4321-213
* **Rust toolchain version:** 1.57.0 (f1edd0429 2021-11-29) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.3 (CL-213.5744.254)
* **Operating system:** Linux 5.15.6-arch2-1
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

I use the `#[inherent]` proc-macro from the [inherent](https://docs.rs/inherent/1.0.0/inherent/) crate.
With this, you can declare function inside an `impl Trait` bloc as `pub`, which would normally be an error.

## Steps to reproduce
```rust
use inherent::inherent;

trait Trait {
    fn f(self);
}

pub struct Struct;

#[inherent]
impl Trait for Struct {
    pub fn f(self) {} // PUB IS IMPORTANT!
}
```

`rustc` compiles this code successfully, but the plugin shows an error:

![image](https://user-images.githubusercontent.com/12938023/147852386-5849643e-7335-4983-8625-df5bbe078335.png)

**Comment by vlad20012** [comment]
Try to enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` experimental feature via the corresponding dialog. To open the dialog, just invoke `Help | Find Action` action and type `Experimental features`
